src/clp_ffi_py/ir/native/PyKeyValuePairLogEvent.cpp

```diff
@@ -0,0 +1,340 @@
+            py_utils_serialize_dict_to_msgpack(py_dict)
+    };
+    if (nullptr == serialized_msgpack_byte_sequence) {
+        return std::nullopt;
+    }
+
+    // Since the type is already checked, we can use the macro to avoid duplicated type checking.
+    std::span<char const> const data_view{
+            PyBytes_AS_STRING(serialized_msgpack_byte_sequence.get()),
+            static_cast<size_t>(PyBytes_GET_SIZE(serialized_msgpack_byte_sequence.get()))
+    };
+    auto const unpack_result{unpack_msgpack(data_view)};
+    if (unpack_result.has_error()) {
+        PyErr_SetString(PyExc_RuntimeError, unpack_result.error().c_str());
+        return std::nullopt;
+    }
+    auto const& msgpack_obj{unpack_result.value().get()};
+    if (msgpack::type::MAP != msgpack_obj.type) {
+        PyErr_SetString(PyExc_TypeError, "Unpacked msgpack is not a map");
+        return std::nullopt;
+    }
+
+    auto serializer_result{
+            clp::ffi::ir_stream::Serializer<clp::ir::four_byte_encoded_variable_t>::create()
+    };
+    if (serializer_result.has_error()) {
+        PyErr_Format(
+                PyExc_RuntimeError,
+                cSerializerCreateErrorFormatStr.data(),
+                serializer_result.error().message().c_str()
+        );
+        return std::nullopt;
+    }
+
+    auto& serializer{serializer_result.value()};
+    if (false == serializer.serialize_msgpack_map(msgpack_obj.via.map)) {
+        PyErr_SetString(PyExc_RuntimeError, cSerializerSerializeMsgpackMapError.data());
+        return std::nullopt;
+    }
+
+    auto const ir_buf{serializer.get_ir_buf_view()};
+    clp::BufferReader buf_reader{
+            clp::size_checked_pointer_cast<char const>(ir_buf.data()),
+            ir_buf.size()
+    };
+
+    auto deserializer_result{
+            clp::ffi::ir_stream::Deserializer<IrUnitHandler>::create(buf_reader, IrUnitHandler{})
+    };
+    if (deserializer_result.has_error()) {
+        PyErr_Format(
+                PyExc_RuntimeError,
+                cDeserializerCreateErrorFormatStr.data(),
+                deserializer_result.error().message().c_str()
+        );
+        return std::nullopt;
+    }
+
+    auto& deserializer{deserializer_result.value()};
+    auto& ir_unit_handler{deserializer.get_ir_unit_handler()};
+    while (true) {
+        auto const result{deserializer.deserialize_next_ir_unit(buf_reader)};
+        if (result.has_error()) {
+            PyErr_Format(
+                    PyExc_RuntimeError,
+                    cDeserializerDeserializeNextIrUnitErrorFormatStr.data(),
+                    result.error().message().c_str()
+            );
+            return std::nullopt;
+        }
+        auto const ir_unit_type{result.value()};
+        if (clp::ffi::ir_stream::IrUnitType::LogEvent == ir_unit_type) {
+            break;
+        }
+        if (clp::ffi::ir_stream::IrUnitType::SchemaTreeNodeInsertion != ir_unit_type) {
+            continue;
```

Comment:
not sure why this `continue` is here, maybe git messed something up?

Reply:
Yeah it's my bad... sorry, removed